armi/nuclearDataIO/cccc.py

```diff
@@ -689,6 +706,57 @@ def _write(cls, lib, fileName, fileMode):
         raise NotImplementedError()
 
 
+class DataContainer:
+    """
+    Data representation that can be read/written to/from with a Stream.
+
+    This is an optional convenience class expected to be used in
+    concert with :py:class:`StreamWithDataStructure`.
+    """
+
+    def __init__(self):
+        # Need Metadata subclass for default keys
+        self.metadata = nuclearFileMetadata._Metadata()
+
+
+class StreamWithDataContainer(Stream):
+    """
+    A stream that reads/writes to a specialized data container.
+
+    This is a relatively common pattern so some of the boilerplate
+    is handled here.
```

Comment:
Might be worth discussing why not all CCCC.Streams use this pattern. Havent refactored yet?